pkg/apis/core/validation/cloudprofile.go

```diff
@@ -101,8 +98,29 @@ func validateKubernetesSettings(kubernetes core.KubernetesSettings, fldPath *fie
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("versions[]").Child("expirationDate"), latestKubernetesVersion.ExpirationDate, fmt.Sprintf("expiration date of latest kubernetes version ('%s') must not be set", latestKubernetesVersion.Version)))
 	}
 
-	versionsFound := sets.New[string]()
+	allErrs = append(allErrs, validateSpecStaticKubernetesVersions(kubernetes.Versions, fldPath)...)
+
 	for i, version := range kubernetes.Versions {
+		idxPath := fldPath.Child("versions").Index(i)
+		allErrs = append(allErrs, validateExpirableVersionInContext(version, kubernetes.Versions, idxPath)...)
+	}
+
+	return allErrs
+}
+
+// k8sVersionCPRegex is used to validate kubernetes versions in a cloud profile.
+var k8sVersionCPRegex = regexp.MustCompile(`^([0-9]+\.){2}[0-9]+$`)
+
+var supportedVersionClassifications = sets.New(string(core.ClassificationPreview), string(core.ClassificationSupported), string(core.ClassificationDeprecated))
+
+func validateSpecStaticKubernetesVersions(versions []core.ExpirableVersion, fldPath *field.Path) field.ErrorList {
```

Comment:
```suggestion
func validateKubernetesVersions(versions []core.ExpirableVersion, fldPath *field.Path) field.ErrorList {
```

Didn't understand "static" in this context 😃 

Does it make sense to move this function to `core/validation/utils.go` the make the re-usability more evident?